src/lib/add-on/clients/add-on_auto.rb

```diff
@@ -1,328 +1,319 @@
-# encoding: utf-8
-
-# File:
-#      add-on_auto.ycp
-#
-# Module:
-#      Add-On autoinstallation and configuration
-#
-# Summary:
-#      Add-On autoinstallation preparation
-#
-# Authors:
-#      Jiri Srain <jsrain@suse.cz>
-#
-# $Id$
-#
-module Yast
-  class AddOnAutoClient < Client
-    def main
-      Yast.import "Pkg"
-      Yast.import "UI"
-      textdomain "add-on"
-
-      Builtins.y2milestone("----------------------------------------")
-      Builtins.y2milestone("add-on auto started")
-
-      Yast.import "AddOnProduct"
-      Yast.import "Progress"
-      Yast.import "AutoinstSoftware"
-      Yast.import "PackageCallbacks"
-      Yast.import "Label"
-      Yast.import "AutoinstGeneral"
-      Yast.import "PackageLock"
-      Yast.import "Installation"
-      Yast.import "String"
+# rubocop:disable Naming/FileName
+
+require "yast"
+require "installation/auto_client"
+
+Yast.import "AutoInstall"
+Yast.import "AddOnProduct"
+Yast.import "Progress"
 
+module Yast
+  class AddOnAutoClient < ::Installation::AutoClient
+    def run
       Yast.include self, "add-on/add-on-workflow.rb"
 
-      @progress_orig = Progress.set(false)
+      progress_orig = Yast::Progress.set(false)
+      ret = super
+      Yast::Progress.set(progress_orig)
+
+      ret
+    end
 
+    def import(data)
+      add_on_products = data.fetch("add_on_products", [])
 
-      @ret = nil
-      @func = ""
-      @param = {}
+      valid_add_on_products = add_on_products.reject.with_index(1) do |add_on, index|
+        reject = false
 
-      # Check arguments
-      if Ops.greater_than(Builtins.size(WFM.Args), 0) &&
-          Ops.is_string?(WFM.Args(0))
-        @func = Convert.to_string(WFM.Args(0))
-        if Ops.greater_than(Builtins.size(WFM.Args), 1) &&
-            Ops.is_map?(WFM.Args(1))
-          @param = Convert.to_map(WFM.Args(1))
+        if add_on.fetch("media_url", "").empty?
+          log.error "Missing <media_url> value in the #{index}. add-on-product definition"
+
+          # abort import/installation
+          return false unless skip_add_on_and_continue?(index)
+
+          reject = true
         end
+
+        reject
```

Comment:
why not having something like this in block?
```
next false unless add_on.fetch("media_url", "").empty?

log.error "Missing <media_url> value in the #{index}. add-on-product definition"

# abort import/installation
return false unless skip_add_on_and_continue?(index)

true
```

Reply:
Oh, nice!

Thank you!